llvm/lib/Transforms/IPO/MergeFunctions.cpp

```diff
@@ -467,9 +488,17 @@ bool MergeFunctions::runOnModule(Module &M) {
   return Changed;
 }
 
+DenseMap<Function *, Function *>
+MergeFunctions::runOnFunctions(ArrayRef<Function *> F) {
+  bool MergeResult = this->run(F);
+  if (!MergeResult)
+    this->DelToNewMap = DenseMap<Function *, Function *>();
```

Comment:
Can this be
```suggestion
  assert(MergeResult == !DelToNewMap.empty());
```
to ensure consistency?

Reply:
Sure!

Resolved in [07aa7c1](https://github.com/llvm/llvm-project/pull/111045/commits/07aa7c1d29fafab9b89401349f20ccb9007b5733).